app/ts/background/background.ts

```diff
@@ -134,17 +135,37 @@ export const simulateGovernanceContractExecution = async (pendingTransaction: Pe
 export const simulateGnosisSafeMetaTransaction = async (gnosisSafeMessage: VisualizedPersonalSignRequestSafeTx, simulationState: SimulationState | undefined, ethereumClientService: EthereumClientService, tokenPriceService: TokenPriceService): Promise<DistributiveOmit<SimulateExecutionReplyData, 'transactionOrMessageIdentifier'>> => {
 	const returnError = (errorMessage: string) => ({ success: false as const, errorType: 'Other' as const, errorMessage })
 	try {
-		const transactionWithoutGas = {
+		const delegateCallExecuteInterface = new ethers.Interface(['function delegateCallExecute(address, bytes memory) payable external returns (bytes memory)'])
+
+		// Call: 0x0, DelegateCall: 0x1
+		// https://github.com/safe-global/safe-smart-account/blob/main/contracts/libraries/Enum.sol
+		const isDelegateCall = gnosisSafeMessage.message.message.operation === 0x1n
+		const ORIGINAL_GNOSIS_SAFE = 0x0000000000000000000000000000000000920515n
+		/*
+		If we are doing a normal call, we send a transaction from gnosis safe to the callable address
+		If we are doing a delegate call, we do a following operation:
+			1) move safe (gnosisSafeMessage.verifyingContract.address) -> ORIGINAL_GNOSIS_SAFE
+			2) replace safe with GnosisSafeProxyProxy (a contract that delegates everything to ORIGINAL_GNOSIS_SAFE, except calls to `delegateCallExecute`)
```

Comment:
I think you can get rid of step 1.  Just replace the existing proxy with our proxy, which does everything the original proxy did *except* has a handler for `delegateCallExecute(...)`.